seagoat/sources/chroma.py

```diff
@@ -87,28 +142,138 @@ def fetch(query_text: str, limit: int):
+            return
+        
+        # Get current batch data
+        ids = batch_buffer.ids.copy()
+        documents = batch_buffer.documents.copy()
+        metadatas = batch_buffer.metadatas.copy()
+        
+        # Clear buffer immediately to avoid re-processing
+        batch_buffer.clear()
+        
+        # Try to add chunks
+        failed_indices = _add_chunks_to_collection(ids, documents, metadatas)
+        
+        # Store failed chunks for potential retry
+        for idx in failed_indices:
+            failed_chunks.append({
+                'id': ids[idx],
+                'document': documents[idx],
+                'metadata': metadatas[idx]
+            })
+        
+        if failed_indices:
+            logger.warning(f"Failed to add {len(failed_indices)} chunks in batch")
+
+    def cache_chunk(chunk):
+        if not batch_buffer:
+            # No batching - use original behavior with error handling
+            try:
+                chroma_collection.add(
+                    ids=[chunk.chunk_id],
+                    documents=[chunk.chunk],
+                    metadatas=[
+                        {
+                            "path": chunk.path,
+                            "line": chunk.codeline,
+                            "git_object_id": chunk.object_id,
+                        }
+                    ],
+                )
+            except IDAlreadyExistsError:
+                logger.debug(f"Chunk {chunk.chunk_id} already exists")
+            except Exception as e:
+                logger.error(f"Failed to cache chunk {chunk.chunk_id}: {e}")
+                failed_chunks.append({
+                    'id': chunk.chunk_id,
+                    'document': chunk.chunk,
+                    'metadata': {
                         "path": chunk.path,
                         "line": chunk.codeline,
                         "git_object_id": chunk.object_id,
                     }
-                ],
+                })
+        else:
+            # Batching enabled
+            should_flush = batch_buffer.add(
+                chunk.chunk_id,
+                chunk.chunk,
+                {
+                    "path": chunk.path,
+                    "line": chunk.codeline,
+                    "git_object_id": chunk.object_id,
+                }
             )
-        except IDAlreadyExistsError:
-            pass
+            
+            if should_flush:
+                flush_batch()
 
     def cache_repo():
-        # chromadb does not need any repo cache action
-        pass
+        """Called at the end of repository processing"""
+        # Flush any remaining batched chunks
+        if batch_buffer:
+            flush_batch()
+        
+        # Log summary of any failed chunks
+        if failed_chunks:
+            logger.warning(f"Total {len(failed_chunks)} chunks failed to cache")
+            # Optionally, we could retry failed chunks here with exponential backoff
```

Comment:
I never observed any "flaky" behavior from ChromaDB, it seems like if a chunk could not be added that is usually because it already exists. I would expect that we won't recover from any errors by adding something more sophisticated